Automattic/woocommerce-payments
# Refactor WC_Payment_Gateway_WCPay Part 1

#### Changes proposed in this Pull Request
Related to #823 
This PR is the first part of a series of refactors of the WC_Payment_Gateway_WCPay class.

#### Testing instructions
- With test mode active
- Go to WooCommerce > Payments
- Make sure the notice `Test mode active: All transactions are simulated. Customers can't make real purchases through WooCommerce Payments.` shows up

- Go to plugins
- Make sure the Settings link shows up and works

- Make [is_available_for_current_currency](https://github.com/Automattic/woocommerce-payments/blob/3d81b3467672c989c9143a16673adb4c4b0d72b1/includes/class-wc-payment-gateway-wcpay.php#L504) return false
- Make sure the `Unsupported currency: The selected currency is not available for the country set in your WooCommerce Payments account.` notice shows up on wp-admin

- As a shopper add a product to the cart
- Make sure you can see the card inputs
- Test again using UPE

- As a shopper buy any product
- As a shopper buy any product using WooPay


-------------------


**Post merge**
